### ISSUE TITLE
Date$ different format

### ISSUE THREAD

**Issue by KarlSevenSeven** [WaynePhillipsEA/twinbasic#846]
**Describe the bug**
Debug.Print Date$ is different in VB6

```
VB6:
05-17-2022

tB:
17.05.2022
```

**To Reproduce**
1. Run `Debug.Print Date$` 

**Expected behavior**
Same output.
Date$ should not respect regional settings.

**Desktop (please complete the following information):**
 - OS: Windows 10
 - twinBASIC beta 40

**Comment by wqweto** [comment]
https://docs.microsoft.com/en-us/office/vba/language/reference/user-interface-help/date-function

At least I learned another VBx trivia/feature I'm never going to use i.e. `VBA.Calendar` property with valid values being `vbCalGreg` and `vbCalHijri` affect the result of `Date$` property (not function) namely:

> If the calendar is Hijri, Date$ returns a 10-character string of the form mm-dd-yyyy, where mm (01–12), dd (01–30) and yyyy (1400–1523) are the Hijri month, day, and year.

Quite amusing :-))

**Comment by EduardoVB** [comment]
It seems to affect both:

```
Private Sub Form_Load()
    Debug.Print Date, Date$
    Calendar = vbCalHijri
    Debug.Print Date, Date$
End Sub
```

in VB6, returns here:
```
17/5/2022     05-17-2022
16/10/1443    10-16-1443
```

**Comment by EduardoVB** [comment]
I found that there is also `Time$`, it seems to always return the same as `Time`.
And that changing the configuration at the control panel does not affect `Time` nor `Time$`, but it does to `FormatDateTime(Now, vbLongTime)` and `FormatDateTime(Now, vbShortTime)`

**Comment by bclothier** [comment]
I think that is different from how VBA behave. It's been long time but I seem to recall that regional setting effects VBA's `Date` & `Time` so.

**Comment by EduardoVB** [comment]
In my test they work exactly the same (Word VBA and VB6).

**Comment by bclothier** [comment]
I just tested --- Windows regional setting has 4 settings but VBA will only use the short date and long time settings in the formatting. Any changes to the long date and short time are ignored.

**Comment by EduardoVB** [comment]
`Time` function does not use the regional setting (as `Date` does). It is invariant.

**Comment by bclothier** [comment]
I double checked but I still see different results between `Time` and `Time$`; former reacts to changes in the long time setting; latter is always `HH:mm:ss`. You said both `Time` and `Time$` are same for you but it's not the case here.

**Comment by WaynePhillipsEA** [comment]
Here are my findings;

```
Date        -> affected by OS Short Date
Time        -> affected by OS Long Time
vbShortDate -> affected by OS Short Date
vbLongDate  -> affected by OS Long Date
vbLongTime  -> affected by OS Long Time

Time$       -> unaffected by OS
Date$       -> unaffected by OS 
vbShortTime -> unaffected by OS
```

**Comment by WaynePhillipsEA** [comment]
So accordingly, `Time$` and `Date$` are currently incorrect in tB, as they are being affected by the OS settings.

**Comment by EduardoVB** [comment]
Guys, did you actually changed the long time format setting at the control panel to test, or just changed the date format and assumed that time would be the same?

I insist, date format is reflected but NOT time.

```
Private Sub Form_Load()
    Debug.Print Date, Time, FormatDateTime(Now, vbLongTime)
End Sub
```
The result, changing date from d/M/aaaa (that would translate to d/M/yyyy) to aaaa/d/M (yyyy/d/M) and time from HH:mm:ss to ss:HH:mm:

```
18/5/2022     09:19:01      09:19:01
2022/18/5     09:19:56      56:09:19
```

Windows 11 21H2
VB6 SP6

**Comment by WaynePhillipsEA** [comment]
> Guys, did you actually changed the long time format setting at the control panel to test, or just changed the date format and assumed that time would be the same?
> 
> I insist, date format is reflected but NOT time.

Of course.  Both @bclothier and I reported that Time DOES reflect our regional settings.  I even double checked it, because of the discrepancy reported.   This was on Win 10 and VB6 (and VBA).

So I will look into it further when I get chance to find out why there's a difference here.

**Comment by bclothier** [comment]
From Windows 10. Note that in 8.1 & Later there are 2 different places to modify the regional settings.
![image](https://user-images.githubusercontent.com/2367644/169049418-1a91249f-a8ca-48f9-a311-5e22bb6b8078.png)

![image](https://user-images.githubusercontent.com/2367644/169049502-854cd315-be22-441d-a2b7-1f477aeffa75.png)

![image](https://user-images.githubusercontent.com/2367644/169049539-7453c4e0-e191-4113-8a80-bc649c9101d3.png)

**Comment by EduardoVB** [comment]
I found that some things are changed but not others. The format whether to put leading zeroes is changed, also the intermediate characters, but not the order of hour, minutes and seconds (that was what I changed in my testings).

So, for example, `ss:::H:mm` actually returns H:::mm:ss at `Time` function.
It is not the same in FormatDateTime function that follow the format strictly.

Weird, not sure if that is a bug or something on purpose.

**Comment by WaynePhillipsEA** [comment]
@EduardoVB could you confirm if the tB behaviour for the `Time` function matches the VBx version for you?

**Comment by WaynePhillipsEA** [comment]
`Date$` and `Time$` should now be correct as of BETA 41.  I'll leave this open until we've confirmed/sorted the `Time` discrepancy.

**Comment by EduardoVB** [comment]
> @EduardoVB could you confirm if the tB behaviour for the `Time` function matches the VBx version for you?

Yes, I performed several tests and the results matched.
Also, that the first separator is used as the second separator too, and the second separator is ignored.
The `FormatDateTime(Now, vbLongTime)` does not do that but follow the format exactly, and that also matched.  👍👌

**Comment by WaynePhillipsEA** [comment]
Great, thanks for confirming.  I'll close this issue then.  Thanks everybody.
